util/session/sessionmanager.go

```diff
@@ -134,16 +135,20 @@ func (mgr *SessionManager) Parse(tokenString string) (jwt.Claims, error) {
 
 // VerifyUsernamePassword verifies if a username/password combo is correct
 func (mgr *SessionManager) VerifyUsernamePassword(username, password string) error {
+	settings, err := mgr.settingsMgr.GetSettings()
```

Comment:
Instead of using settings manager, to check if admin is disabled, the `adminDisable` setting can be stored in the `SettingManage` data structure and passed via NewSessionManager at https://github.com/argoproj/argo-cd/blob/57ea24281c509aa47d575f1478892f46ed0a704c/server/server.go#L178.